src/lib/pytch/syscalls.js

```diff
@@ -3,6 +3,69 @@ var $builtinmodule = function (name) {
 
     const skulpt_function = Sk.pytchsupport.skulpt_function;
 
+    const validKeys = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", 
+    "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", 
+    "0", "1", "2", "3", "4", "5", "6", "7", "8", "9",
+    " ", "ArrowLeft", "ArrowDown", "ArrowUp", "ArrowRight"];
+
+    const map = {};
+
+    validKeys.forEach(validKey => {
+        if (validKey !== " ")
+            map[validKey.toLowerCase()] = validKey;
+    });
+
+    const isValidKeyname = (keyname) => validKeys.includes(keyname)
+    
+    function suggestedKeyname(invalidKeyname) {
+        //Remove symbols and capital letters to suggest a keyname
+        //Eg "Arrowleft", "arrowLeft" or "Arrow-Left" -> "ArrowLeft"
+        const invalidKeynameLower = invalidKeyname.toLowerCase();
+        var invalidKeynameCleaned = invalidKeynameLower.trim().replace(/[^a-z0-9]/g, "");
+        const suggested_key = map[invalidKeynameCleaned];
+        return suggested_key || null;
+    }
+
+    mod._assert_keyname_valid = skulpt_function(
+        (keyname) => {
+            assertPyKeynameValid(keyname);
+        },
+        `(KEY_NAME) throws error if keyname is not valid; continue otherwise`,
+    );
+
+    function assertPyKeynameValid(py_keyname) {
+        if (!Sk.builtin.checkString(py_keyname))
+            throw new Sk.builtin.ValueError(
+                "keyname must be a string"
+            );
+        jsKeyname = py_keyname.v;
+        if (isValidKeyname(jsKeyname))
+            return;
+        if (!jsKeyname)
+            throw new Sk.builtin.ValueError(
+                `keyname must not be an empty string.`
+            );
+        if (!jsKeyname.trim())
+            throw new Sk.builtin.ValueError(
+                `keyname must be a valid key; `
+                + `if you meant the spacebar, use " " (that's a string `
+                + `consisting of a single space character)`
+            );
+        maybe_suggestion = suggestedKeyname(jsKeyname)
+        if (maybe_suggestion != null)
+            throw new Sk.builtin.ValueError(
+                `keyname must be a valid key. `
+                +`Maybe you meant "${maybe_suggestion}"`
+            );
```

Comment:
The error messages should be consistently punctuated.  I prefer the form

> keyname must be a valid key; if you meant the spacebar...

(also with no terminating full stop), rather than

> keyname must be a valid key.  Maybe you meant...